src/object/srv_obj_migrate.c

```diff
@@ -590,9 +599,17 @@ migrate_pool_tls_lookup_create(struct ds_pool *pool, unsigned int version, unsig
 
 		return rc;
 	}
-
-	d_getenv_uint(ENV_MIGRATE_ULT_CNT, &max_migrate_ult);
 	D_ASSERT(generation != (unsigned int)(-1));
+
+	arg.max_ult_cnt = MIGRATE_DEFAULT_MAX_ULT;
+	d_getenv_uint(ENV_MIGRATE_ULT_CNT, &arg.max_ult_cnt);
+	if (arg.max_ult_cnt < MIGRATE_MIN_MAX_SIZE)
+		arg.max_ult_cnt = MIGRATE_MIN_MAX_SIZE;
+	arg.max_mem_size = MIGRATE_DEFAULT_MAX_SIZE;
+	if (arg.max_mem_size < MIGRATE_MIN_MAX_ULT)
+		arg.max_mem_size = MIGRATE_MIN_MAX_ULT;
+	d_getenv_uint64_t(ENV_MIGRATE_MEM_SIZE, &arg.max_mem_size);
```

Comment:
looks like it is mis-ordered? should first get the env, then compare with min?

BTW, the "MIGRATE_MIN_MAX_ULT 32" looks too small, in that case if with 16 tgts, then each VOS can only create 2 ULT for migration? if with 32 tgts then can only with 1ULT per tgt that even cannot work. Maybe the min ult count could be 512?
the "MIGRATE_MIN_MAX_SIZE" 1MB looks also too small, but it is not a hard limit although.

Reply:
maybe change not committed before push?
the "d_getenv_uint64_t(ENV_MIGRATE_MEM_SIZE, &arg.max_mem_size);" still mis-ordered, should be moved to L611?